pkg/segment/structs/evaluationstructs.go

```diff
@@ -717,50 +733,77 @@ func (self *BoolExpr) Evaluate(fieldToValue map[string]utils.CValueEnclosure) (b
 		isRightValNull := toputils.IsNilValueError(errRight) || rightVal == nil
 
 		if (errLeft != nil && !isLeftValNull) || (errRight != nil && !isRightValNull) {
-			return false, fmt.Errorf("BoolExpr.Evaluate: error evaluating ValueExprs, errLeft: %v, errRight: %v", errLeft, errRight)
+			return nil, fmt.Errorf("BoolExpr.Evaluate: error evaluating ValueExprs, errLeft: %v, errRight: %v", errLeft, errRight)
 		}
 
 		switch self.ValueOp {
 		case "=", "!=":
 			if isLeftValNull || isRightValNull {
-				if isLeftValNull && isRightValNull {
-					return self.ValueOp == "=", nil
+				if isLeftValNull && isRightValNull && self.ValueOp == "=" {
+					// if both values are null, then the evaluation should result in true for "="
+					return getBoolCValueEnclosure(true), nil
+				} else if self.ValueOp == "!=" {
+					// if either value is null, then the evaluation should result in true for "!="
+					return getBoolCValueEnclosure(true), nil
 				}
-				return self.ValueOp == "!=", nil
+
+				// if either value is null, then the evaluation should result in false for "="
+				// But since one of the values is null, the evaluation should send a nil value
+				return &utils.CValueEnclosure{Dtype: utils.SS_DT_BACKFILL, CVal: nil}, nil
```

Comment:
I'm confused. You specify a situation where we should return `false`, but then you always return null?

Reply:
Yes, i have updated the code, to actually return `NULL` if left or right value is `null`. As that is how splunk is executing.